src/serv_account/errors.rs

```diff
@@ -13,13 +15,22 @@ pub enum ServiceAccountError {
     Base64Decode(#[from] base64::DecodeError),
 
     #[error("failed to create rsa key pair: {0}")]
-    RsaKeyPair(String),
+    RsaKeyPair(KeyRejected),
 
     #[error("failed to rsa sign: {0}")]
-    RsaSign(String),
+    RsaSign(Unspecified),
 
     #[error("failed to send request")]
-    HttpReqwest(#[from] reqwest::Error),
+    HttpRequest(reqwest::Error),
+
+    #[error("failed to send request")]
+    HttpRequestUnsuccessful(StatusCode, std::result::Result<String, reqwest::Error>),
```

Comment:
std::result::Result could be shorted to StdResult, since there is already an import on line 3